privacyidea/lib/tokens/webauthntoken.py

```diff
@@ -1069,13 +1069,30 @@ def create_challenge(self, transactionid=None, options=None):
 
         message = self._get_message(options)
 
-        # To aid with unit testing a fixed nonce may be passed in.
-        nonce = options['nonce'] if 'nonce' in options else self._get_nonce()
+        # if a transaction id is given, check if there are other webauthn
+        # token and reuse the challenge.
+        # TODO: It might be more sensible to pass around a list of all tokens
+        #  currently doing challenge creation in this request.
+        challenge = None
+        if transactionid:
+            for c in get_challenges(transaction_id=transactionid):
+                # TODO: this throws an exception if the token does not exists
+                #  but just created a challenge with it...
+                if get_tokens(serial=c.serial, tokentype=self.get_class_type(),
+                              count=True):
+                    challenge = c.challenge
+                    break
+
+        if not challenge:
+            nonce = self._get_nonce()
+            challenge = hexlify_and_unicode(nonce)
+        else:
+            nonce = binascii.unhexlify(challenge)
 
         # Create the challenge in the database
         challenge = Challenge(serial=self.token.serial,
```

Comment:
can you please rename this to "db_challenge", so that it looks the same like in u2ftoken?

Otherwise we have a "challenge", that is the hexlified nonce and here it gets the challenge object in the DB.

Reply:
will do

Reply:
Done in ecf651a6